buildconfig/Jenkins/Conda/conda-build-recipes

```diff
@@ -0,0 +1,121 @@
+#
+# Example command to build all of the packages:
+# conda-build-recipes $WORKSPACE --build-mantid --build-qt --build-workbench
+#
+# Expected args:
+#   1. WORKSPACE: path to the workspace/source code that this should run inside
+#                 (mantid repo). Windows Caveat: Only use / for this arguement do
+#                 not use \\ or \ in the path.
+#
+# Possible flags:
+#   --build-mantid: build the mantid package using conda-build
+#   --build-qt: build the qt package using conda-build
+#   --build-workbench: build the workbench package using conda-build
+#
+# Possible parameters:
+#   --clobber-yml: clobber the meta yaml files of each package build with conda-build, provide an absolute path to the uteyml file
+#
+
+# Setup expected variables
+WORKSPACE=$1
+shift
+if [[ $OSTYPE == "msys"* ]]; then
+    WORKSPACE=${WORKSPACE////\\}
+    # Require git settings so that paths can be checked out with longer file names than 260 characters
+    if [[ ! $(git config --system --get core.longpaths) == true ]]; then
+        echo "This windows machine is not configured properly, please run this command as an admin: git config --system core.longpaths true"
+        exit 1
+    fi
+fi
+EXPECTED_MAMBAFORGE_PATH=$WORKSPACE/mambaforge # Install into the WORKSPACE_DIR
+if [[ $OSTYPE == "msys" ]]; then
+    EXPECTED_CONDA_PATH=$EXPECTED_MAMBAFORGE_PATH/condabin/mamba.bat
+else
+    EXPECTED_CONDA_PATH=$EXPECTED_MAMBAFORGE_PATH/bin/mamba
+fi
+CONDA_ENV_NAME=mantid-conda-build-env
+RECIPES_DIR=$WORKSPACE/conda-recipes
+SCRIPT_DIR=$WORKSPACE/buildconfig/Jenkins/Conda/
+
+# Parse flags
+ENABLE_BUILD_MANTID=false
+ENABLE_BUILD_QT=false
+ENABLE_BUILD_WORKBENCH=false
+
+# Handle flag inputs
+while [ ! $# -eq 0 ]
+do
+    case "$1" in
+        --build-mantid) ENABLE_BUILD_MANTID=true ;;
+        --build-qt) ENABLE_BUILD_QT=true ;;
+        --build-workbench) ENABLE_BUILD_WORKBENCH=true ;;
+        --clobber-yml)
+            CLOBBER_FILE="--clobber-file $2"
+            shift ;;
+        *)
+            echo "Argument not accepted: $1"
+            exit 1
+            ;;
+  esac
+  shift
+done
+
+# Build dependencies for requested packages
+if [[ $ENABLE_BUILD_QT  == true ]]; then
+    ENABLE_BUILD_MANTID=true
+fi
+
+if [[ $ENABLE_WORKBENCH  == true ]]; then
+    ENABLE_BUILD_MANTID=true
+    ENABLE_BUILD_QT=true
+fi
+
+# Setup Mambaforge
+$SCRIPT_DIR/download-and-install-mambaforge $EXPECTED_MAMBAFORGE_PATH $EXPECTED_CONDA_PATH true
+
+# Remove conda env if it exists
+$EXPECTED_CONDA_PATH env remove -n $CONDA_ENV_NAME
+
+# Create env with conda-verify and boa (Mamba but for conda-build) installed
+$EXPECTED_CONDA_PATH create -n $CONDA_ENV_NAME boa conda-verify -y
+
+# Activate Conda environment
+. $WORKSPACE/mambaforge/etc/profile.d/conda.sh
+conda activate $CONDA_ENV_NAME
+
+# Clone the recipes
+cd $WORKSPACE
+if [[ -d conda-recipes ]]; then
+    rm -rf conda-recipes
+fi
+git clone https://github.com/mantidproject/conda-recipes.git
```

Comment:
Can we get Jenkins to handle this instead, since we can easily ask if to clone depth=1, handle refspecs and cleanup after each build too?
If not can we simply do something like

```bash
if [[ -d conda-recipes ]]; then
    git fetch && git clean -fxd && git reset --h origin/main
else
    git clone https://github.com/mantidproject/conda-recipes.git
fi
```